### PR TITLE
Feat: SocialMediaShareButton and SocialMediaShareDialog 

### LINKED ISSUE #503: social media share button for project page
implement the share button for social media and messengers. Similar to #481 
located at the end of the "tab line": 
image

### PR DESCRIPTION
**Pull request by verena-e-p**
Closes #503

## Description #503
The components SocialMediaShareButton and SocialMediaShareDialog are now implemented and both added to the project page. The SocialMediaShareDialog is integrated into the SocialMediaShareButton component. The dialog contains multiple buttons to share a link on various social media platforms / via e-mail as well as a text field with a link that can be copied to share manually.

## Functionality SocialMediaShareButton
* open the SocialMediaShareDialog **or** (if available) the native share dialog of the used device (e.g. smartphone)
* create a database record (containing the user (if logged in), the project, how it was shared) when 
  * opening the native share dialog

## Functionality SocialMediaShareDialog
* open a new window where the link can be shared when the user clicks on a social media / the mail button
* copy the link to the clipboard when the user clicks on the copy link button
* create a database record (containing the user (if logged in), the project, how it was shared) when 
  * clicking on one of the social media buttons / the mail button
  * clicking on the 'copy link' button (only once per session)

## Before landing

1. PR has meaningful title
1. `yarn lint` passes
1. `yarn format` passes

### CODE REVIEW

**Comment by verena-e-p** [comment]
Is there any way to find out whether the project image has loaded successfully? I know that it is a required field and therefore should not be empty, but I was wondering about error cases that cause the picture to not be displayed. My reasoning behind that is is, that the buttons are displayed weirdly without the image because of their absolute positioning in relation to the image. Using `onError` did not work for me because it did not seem to care whether the URL is correct or not.

File: frontend/public/texts/project_texts.js [hunk 1]
```diff
@@ -923,6 +923,30 @@ export default function getProjectTexts({ project, user, url_slug, locale, creat
...
+      de: "Link",
+    },
+    climate_protection_project_by: {
+      en: "Climate Protection Project by ",
+      de: "Klimaschutzprojekt von  ",
+    },
+    this_is_the_link_to_the_project: {
+      en: 'This is the link to "' + project?.name + '" on Climate Connect: ',
+      de: 'Das ist der Link zu "' + project?.name + '" auf Climate Connect: ',
+    },
```
Comment (positiveimpact) [inline frontend/public/texts/project_texts.js:949]: I'd try to use some more engaging messages here

File: frontend/src/components/shareContent/SocialMediaShareDialog.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  title,
+}) {
+  const classes = useStyles();
+
+  const handleClose = () => {
+    onClose(false);
+  };
+
+  const facebookHashtag = "#believeintogether";
+  const twitterHastags = ["believeintogether"];
```
Comment (positiveimpact) [inline frontend/src/components/shareContent/SocialMediaShareDialog.js:51]: I'd use camel case of hte hashtags (BelieveInTogether)

**Review by positiveimpact — APPROVED** [review]
Really great job on this! LGTM except the share messages. I'd try to make them a bit more engaging. I'll create a commit to change the messages and then merge it :+1:

File: frontend/src/components/shareContent/SocialMediaShareDialog.js (hunk 2 above)
Comment (verena-e-p) [reply frontend/src/components/shareContent/SocialMediaShareDialog.js]: done :+1:

**Pull request merged**